lib/src/collections.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:typesense/src/api_call.dart';
+
+class Collections {
+  ApiCall _apicall;
+  static const String RESOURCEPATH = '/collections';
+
+  Collections(apicall) {
```

Comment:
@HarisaranG since JavaScript is a dynamically typed language, there would be no types involved in the code. But since dart is a statically typed language and being statically typed has its advantages, we'll want to declare types when they cant be inferred. So in this case either
```dart
Collections(ApiCall apicall) {
  this._apicall = apicall;
}
```
or
```dart
Collections(this._apiCall);
```
is desirable.